x-pack/plugins/apm/public/components/app/correlations/correlations_chart.tsx

```diff
@@ -160,82 +164,86 @@ export function CorrelationsChart({
-          tickFormat={(d) => durationFormatter(d).formatted}
-        />
-        <Axis
-          id="y-axis"
-          domain={yAxisDomain}
-          title={i18n.translate(
-            'xpack.apm.correlations.latency.chart.numberOfTransactionsLabel',
-            { defaultMessage: '# transactions' }
-          )}
-          position={Position.Left}
-          tickFormat={(d) =>
-            d === 0 || Number.isInteger(Math.log10(d)) ? d : ''
-          }
-        />
-        <AreaSeries
-          id={i18n.translate(
-            'xpack.apm.correlations.latency.chart.overallLatencyDistributionLabel',
-            { defaultMessage: 'Overall latency distribution' }
-          )}
-          xScaleType={ScaleType.Log}
-          yScaleType={ScaleType.Log}
-          data={overallHistogram}
-          curve={CurveType.CURVE_STEP_AFTER}
-          xAccessor="key"
-          yAccessors={['doc_count']}
-          color={euiTheme.eui.euiColorVis1}
-          fit="lookahead"
-        />
-        {Array.isArray(histogram) &&
-          field !== undefined &&
-          value !== undefined && (
-            <AreaSeries
-              id={i18n.translate(
-                'xpack.apm.correlations.latency.chart.selectedTermLatencyDistributionLabel',
-                {
-                  defaultMessage: '{fieldName}:{fieldValue}',
-                  values: {
-                    fieldName: field,
-                    fieldValue: value,
-                  },
-                }
-              )}
-              xScaleType={ScaleType.Log}
-              yScaleType={ScaleType.Log}
-              data={histogram}
-              curve={CurveType.CURVE_STEP_AFTER}
-              xAccessor="key"
-              yAccessors={['doc_count']}
-              color={euiTheme.eui.euiColorVis2}
-            />
-          )}
-      </Chart>
-      <EuiSpacer size="s" />
+        <Chart>
+          <Settings
+            rotation={0}
+            theme={chartTheme}
+            showLegend
+            legendPosition={Position.Bottom}
+          />
+          <LineAnnotation
+            id="annotation_1"
+            domainType={AnnotationDomainType.XDomain}
+            dataValues={annotationsDataValues}
+            style={annotationsStyle}
+            marker={`${markerPercentile}p`}
+            markerPosition={'top'}
+          />
+          <Axis
+            id="x-axis"
+            title=""
+            position={Position.Bottom}
+            tickFormat={(d) => {
+              const unit = getDurationUnitKey(d, 1);
+              const converted = getUnitLabelAndConvertedValue(unit, d);
+              const convertedValueParts = converted.convertedValue.split('.');
+              const convertedValue =
+                convertedValueParts.length === 2 &&
+                convertedValueParts[1] === '0'
+                  ? convertedValueParts[0]
+                  : converted.convertedValue;
+              return `${convertedValue}${converted.unitLabel}`;
```

Comment:
@walterra Noticing this and a couple of other things again, I realise I might have been a little quick when reviewing this first time. 

In general I think we should aim to avoid complexity as much as we can. If the existing abstractions cause things to be more complex we should find other ways

In this case it seems like we are not satisfied with the format returned by `getUnitLabelAndConvertedValue`, which leaves us with a couple of options:
 -  Change the implementation of `getUnitLabelAndConvertedValue` (probably affects a lot of other places so might not be a good idea)
 - Create a new formatter and use that (we want to keep the number of formatters low though)
 - Change our requirements and accept the existing format of `getUnitLabelAndConvertedValue`

Reply:
Thanks for the feedback, sure we can revisit this part! I added an item to the meta issue: https://github.com/elastic/kibana/issues/106381